CapstoneProject/Assets/Scripts/Core/Components/Characters/Playable_Character/PlayerAbilityComponent.cs

```diff
@@ -93,14 +99,64 @@ public GameObject[] ScanForEnemiesInArea()
         }
         return allEnemies.ToArray();
     }
-    public GameObject ScanForLowestHealthEnemy()
+    public Enemy ScanForNearestTarget()
     {
-        //TODO: Need to implement the this function
-        GameObject[] targets = GameObject.FindGameObjectsWithTag("Enemy");
-        GameObject lowestHealthEnemy = targets[0];
-        lowestHealthTarget = lowestHealthEnemy;
+        Collider2D[] allEnemiesCollider = Physics2D.OverlapCircleAll(this.transform.position, viewRadius, _enemyLayerMask);
+
+        List<Enemy> enemies = new List<Enemy>();
+        Enemy nearestEnemy = null;
+        if (allEnemiesCollider.Length > 0)
+        {
+            if (_player.rb.CurrentFacingDirection == RigidbodyComponent.FacingDirections.RIGHT)
+            {
+                foreach (var enemyInRange in allEnemiesCollider)
+                {
+                    if (enemyInRange.transform.position.x > _player.transform.position.x)
+                    {
+                        enemies.Add(enemyInRange.GetComponent<Enemy>());
+                    }
+                }
+            }
+            else
+            {
+                foreach (var enemyInRange in allEnemiesCollider)
+                {
+                    if (enemyInRange.transform.position.x < _player.transform.position.x)
+                    {
+                        enemies.Add(enemyInRange.GetComponent<Enemy>());
+                    }
+                }
+            }
```

Comment:
You do the same loop regardless of this high level if so you should nest this if inside the loop to reduce the code and then you could || the inner if's.

          
                foreach (var enemyInRange in allEnemiesCollider)
                {
                    if ((_player.rb.CurrentFacingDirection == RigidbodyComponent.FacingDirections.RIGHT && enemyInRange.transform.position.x > _player.transform.position.x) || enemyInRange.transform.position.x < _player.transform.position.x)
                        {
                            enemies.Add(enemyInRange.GetComponent<Enemy>());
                        }
                }